src/cow-react/modules/limitOrders/updaters/InitialPriceUpdater/index.tsx

```diff
@@ -0,0 +1,16 @@
+import { useEffect } from 'react'
+import { useUpdateAtom } from 'jotai/utils'
+import { updateLimitRateAtom } from '@cow/modules/limitOrders/state/limitRateAtom'
+import { useGetInitialPrice } from '@cow/modules/limitOrders/hooks/useGetInitialPrice'
+
+// Fetch and update initial price for the selected token pair
+export function InitialPriceUpdater() {
+  const updateLimitRateState = useUpdateAtom(updateLimitRateAtom)
+  const { price, isLoading } = useGetInitialPrice()
```

Comment:
Are you sure this is only executed once for the initial price? `useGetInitialPrice` uses a lot of hooks, and would probably be called a few times, right?

Reply:
The actual `native_price` request happens only once for each currency and then when some currency is changed there is another request for that new currency. And the updater's `useEffect` is called every time a new price or isLoading is changed. But I think I see your point, so I will update the name to `MarketPriceUpdater`